### PR TITLE
Binary blob testing

### PR DESCRIPTION
**Pull request by m-1-k-3**
This is a first attempt to do some pre-checking of the firmware
* is it a directory -> proceed with emba as we know it
* is it a file -> pre-checking

Pre-checking includes new kind of modules in the modules diretory starting with Pxyz:
* modules/P02_firmware_bin_file_check.sh  
* modules/P05_firmware_bin_extractor.sh  
* modules/P07_firmware_bin_base_analyser.sh

We execute these modules on the binary file and do some information gathering like an "ls -l" or a "file" command (P02_firmware_bin_file_check.sh):

image

afterwards we binwalk it (P05_firmware_bin_extractor.sh):

image

The extracted firmware and the png from the entropy analysis is stored in the log directory:

image

and the 3rd new module tries to check if the extracted stuff is LInux or something else:

image

If it is Linux we proceed with emba as we know it.
If it is something else like VxWorks we just show it:

image

with this in place we have the base for some more non-Linux checks ...

### CODE REVIEW

File: modules/P07_firmware_bin_base_analyser.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```
Comment (p4cx) [inline modules/P07_firmware_bin_base_analyser.sh:1]: Filename should be also ...analyzer.sh

**Review by p4cx — COMMENTED** [review]
There is an issue with the firmware architecture checker (it won't check for architecture, if emba get a binary instead of a directory) - I will fix this issue.

**Review by p4cx — CHANGES_REQUESTED** [review]
Please fix all open comments, then it will be merged into master. 
Really nice feature!

File: installer.sh [hunk 2]
```diff
@@ -35,7 +35,14 @@ apt-get install shellcheck -y
 apt-get install device-tree-compiler -y
 apt-get install docker.io -y
 apt-get install unzip -y
-apt-get install qemu-user-static -y
+if ! command -v qemu-mips-static > /dev/null ; then
+  echo -e "\\n""$ORANGE""$BOLD""Install qemu package""$NC"
+  apt-get install qemu-user-static -y
+fi
```
[the remark this answers is not in the document]
Comment (m-1-k-3) [reply installer.sh:41]: but probably it is already installed from source and you don't want to re-install your own version

[the remark this answers is not in the document]
Comment (p4cx) [reply installer.sh:41]: good point :)

**Review by p4cx — APPROVED** [review]

**Pull request merged**